src/containers/FormikForm/VersionActionButtons.tsx

```diff
@@ -29,11 +37,77 @@ const VersionActionButtons = ({
   resetVersion,
   version,
   currentLanguage,
+  canDeleteCurrentRevision,
+  articleChanged,
 }: Props) => {
   const { t } = useTranslation();
+  const { values, initialValues, dirty, isSubmitting } = useFormikContext<ArticleFormType>();
+  const [deleteConfirmationOpen, setDeleteConfirmationOpen] = useState(false);
+  const deleteCurrentRevision = useDeleteCurrentRevision();
+  const [deleteError, setDeleteError] = useState<string>("");
+
+  const formIsDirty = useMemo(
+    () =>
+      isFormikFormDirty({
+        values,
+        initialValues,
+        dirty,
+        changed: articleChanged,
+      }) || isSubmitting,
+    [values, initialValues, dirty, articleChanged, isSubmitting],
+  );
+
+  // TODO: Need to refresh content of Slate editor
+  const onDeleteCurrentRevision = () => {
+    setDeleteError("");
+    deleteCurrentRevision.mutate(
+      { articleId: article.id },
+      {
+        onSuccess: () => setDeleteConfirmationOpen(false),
+        onError: () => setDeleteError(t("errorMessage.genericError")),
+      },
+    );
+  };
+
+  const deleteDialogTitle = t("form.workflow.deleteCurrentRevision.title");
+  const deleteButtonLabel = t("alertDialog.delete");
+
+  if (current && canDeleteCurrentRevision)
+    return (
+      <>
+        <IconButton
+          disabled={formIsDirty}
+          variant="danger"
+          size="small"
+          aria-label={deleteButtonLabel}
+          title={deleteButtonLabel}
+          onClick={() => setDeleteConfirmationOpen(true)}
+        >
```

Comment:
Er det bedre å vise knappen disabla dersom versjonen ikkje kan vises? Må i tilfelle ha en anna title/aria-label med "Kan ikke slette denne revisjonen" eller nokke sånt.
```suggestion
  if (current)
    return (
      <>
        <IconButton
          disabled={formIsDirty || !canDeleteCurrentRevision}
          variant="danger"
          size="small"
          aria-label={deleteButtonLabel}
          title={deleteButtonLabel}
          onClick={() => setDeleteConfirmationOpen(true)}
        >
```